Userland/Libraries/LibWeb/HTML/HTMLTableElement.cpp

```diff
@@ -40,4 +44,71 @@ void HTMLTableElement::apply_presentational_hints(CSS::StyleProperties& style) c
     });
 }
 
+NonnullRefPtr<DOM::HTMLCollection> HTMLTableElement::rows()
+{
+
+    HTMLTableElement* table_node = this;
+    // FIXME:  The elements in the collection must be ordered such that those elements whose parent is a thead are
+    //         included first, in tree order, followed by those elements whose parent is either a table or tbody
+    //         element, again in tree order, followed finally by those elements whose parent is a tfoot element,
+    //         still in tree order.
+    // How do you sort HTMLCollection?
+
+    return DOM::HTMLCollection::create(*this, [table_node](DOM::Element const& element) {
+        // Only match TR elements which are:
+        // * children of the table element
+        // * children of the thead, tbody, or tfoot elements that are themselves children of the table element
+        if (!is<HTMLTableRowElement>(element)) {
+            return false;
+        }
+        if (element.parent_element() == table_node)
+            return true;
+
+        if (element.parent_element() && (element.parent_element()->tag_name() == TagNames::thead || element.parent_element()->tag_name() == TagNames::tbody || element.parent_element()->tag_name() == TagNames::tfoot)
+            && element.parent()->parent() == table_node) {
+            return true;
+        }
+
+        return false;
+    });
+}
+
+DOM::ExceptionOr<NonnullRefPtr<HTMLTableRowElement>> HTMLTableElement::insert_row(long index)
+{
+    if (index < -1 || index >= (long)rows()->length()) {
```

Comment:
We call `rows()` over and over in this function, creating a new `HTMLCollection` object every time. Since we don't currently implement the `SameObject` behavior, we should make sure to only call it once.

Reply:
Fixed, replicated in `delete_row` for consistency :)

Reply:
Fixed, replicated in `delete_row` for consistency :)